### PR TITLE
[tests-only][full-ci]added then steps for apiProvisioning-v2 suite

### PR DESCRIPTION
**Pull request by sushmita56**
## Description
This PR adds ``` Then ``` steps to existing API test scenarios to better check the results of ``` When ``` steps.

### Suite Covered
- ``` apiProvisioning-v2```

## Related Issues
- Part of https://github.com/owncloud/core/issues/39512

## How Has This Been Tested?
- Locally
- CI

## Types of changes
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by phil-davis** [comment]
@sushmita56  I just merged PR #39688 and that added some scenarios to `editUser.feature`

Please check the new scenarios - they also might need to have HTTP status checks etc added to them.

File: tests/acceptance/features/apiProvisioning-v2/enableUser.feature [hunk 1]
```diff
@@ -74,7 +76,7 @@ Feature: enable user
       | Brian    |
     And user "Brian" has been disabled
     When user "Alice" tries to enable user "Brian" using the provisioning API
-    Then the OCS status code should be "401"
+    Then the OCS status code should be "997"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiProvisioning-v2/enableUser.feature:79]: this change may have to be reported if true
cc @phil-davis

File: tests/acceptance/features/apiProvisioning-v2/getUser.feature [hunk 2]
```diff
@@ -31,6 +31,7 @@ Feature: get user
     And the display name returned by the API should be "<displayname>"
     And the email address returned by the API should be "<email>"
     And the quota definition returned by the API should be "default"
+    And the quota definition returned by the API should be "default"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiProvisioning-v2/getUser.feature:34]: duplicated,

File: tests/acceptance/features/apiProvisioning-v2/getUsers.feature [hunk 3]
```diff
@@ -47,6 +47,6 @@ Feature: get users
       | brand-new-user   |
       | another-new-user |
     When user "brand-new-user" gets the list of all users using the provisioning API
-    Then the OCS status code should be "401"
+    Then the OCS status code should be "997"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiProvisioning-v2/getUsers.feature:50]: here too. any idea why the code changed?

Comment (sushmita56) [reply tests/acceptance/features/apiProvisioning-v2/getUsers.feature:50]: It is beacuse the OCS status code for an unauthorized access is `997`.

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

File: tests/acceptance/features/apiProvisioning-v2/resetUserPassword.feature [hunk 4]
```diff
@@ -147,12 +147,7 @@ Feature: reset user password
       | brand-new-user | %regular% | New user    | brand.new.user@oc.com.np |
     And a new browser session for "brand-new-user" has been started
     And the user has generated a new app password named "my-client"
-    When the user "brand-new-user" requests these endpoints with "PROPFIND" to get property "d:getetag" using basic auth and generated app password about user "brand-new-user"
-      | endpoint                                           |
-      | /remote.php/dav/files/%username%/textfile0.txt     |
-    Then the HTTP status code of responses on all endpoints should be "207"
-    When user "brand-new-user" resets the password of user "brand-new-user" to "%alt1%" using the provisioning API
-    Then the content of file "textfile0.txt" for user "brand-new-user" using password "%alt1%" should be "ownCloud test text file 0" plus end-of-line
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiProvisioning-v2/resetUserPassword.feature:155]: why removing these steps?

Comment (sushmita56) [reply tests/acceptance/features/apiProvisioning-v2/resetUserPassword.feature:155]: The second `When` step has been changed into the `Given` step. The first `When` step is making a request before the `brand-new-user` resets his login password.  And the test scenario for this case has been added in the `apiProvisioningUsingAppPassword.feature` file. Also here we're testing whether the app's password is preserved or not after resetting the login password. So in my opinion these steps can be excluded from this scenario.

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by individual-it — APPROVED** [review]

**Pull request merged**